builtin/providers/google/resource_container_cluster.go

```diff
@@ -0,0 +1,445 @@
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		cluster.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("logging_service"); ok {
+		cluster.LoggingService = v.(string)
+	}
+
+	if v, ok := d.GetOk("monitoring_service"); ok {
+		cluster.MonitoringService = v.(string)
+	}
+
+	if v, ok := d.GetOk("network"); ok {
+		cluster.Network = v.(string)
+	}
+
+	if v, ok := d.GetOk("node_config"); ok {
+		nodeConfigs := v.([]interface{})
+		if len(nodeConfigs) > 1 {
+			return fmt.Errorf("Cannot specify more than one node_config.")
+		}
+		nodeConfig := nodeConfigs[0].(map[string]interface{})
+
+		cluster.NodeConfig = &container.NodeConfig{}
+
+		if v, ok = nodeConfig["machine_type"]; ok {
+			cluster.NodeConfig.MachineType = v.(string)
+		}
+
+		if v, ok = nodeConfig["disk_size_in_gb"]; ok {
+			cluster.NodeConfig.DiskSizeGb = v.(int64)
+		}
+
+		if v, ok := nodeConfig["oauth_scopes"]; ok {
+			scopesList := v.([]interface{})
+			scopes := []string{}
+			for _, v := range scopesList {
+				scopes = append(scopes, v.(string))
+			}
+
+			cluster.NodeConfig.OauthScopes = scopes
+		}
+	}
+
+	req := &container.CreateClusterRequest{
+		Cluster: cluster,
+	}
+
+	op, err := config.clientContainer.Projects.Zones.Clusters.Create(
+		config.Project, zoneName, req).Do()
+	if err != nil {
+		return err
+	}
+
+	// Wait until it's created
+	wait := resource.StateChangeConf{
+		Pending:    []string{"PENDING", "RUNNING"},
+		Target:     "DONE",
+		Timeout:    30 * time.Minute,
+		MinTimeout: 3 * time.Second,
+		Refresh: func() (interface{}, string, error) {
+			resp, err := config.clientContainer.Projects.Zones.Operations.Get(
+				config.Project, zoneName, op.Name).Do()
+			log.Printf("[DEBUG] Progress of creating GKE cluster %s: %s",
+				clusterName, resp.Status)
+			return resp, resp.Status, err
+		},
+	}
+
+	_, err = wait.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] GKE cluster %s has been created", clusterName)
+
+	d.SetId(clusterName)
+
+	return resourceContainerClusterRead(d, meta)
+}
+
+func resourceContainerClusterRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	zoneName := d.Get("zone").(string)
+
+	cluster, err := config.clientContainer.Projects.Zones.Clusters.Get(
+		config.Project, zoneName, d.Get("name").(string)).Do()
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", cluster.Name)
+	d.Set("zone", cluster.Zone)
+	d.Set("endpoint", cluster.Endpoint)
+
+	masterAuth := []map[string]interface{}{
+		map[string]interface{}{
+			"username":               cluster.MasterAuth.Username,
+			"password":               cluster.MasterAuth.Password,
+			"client_certificate":     cluster.MasterAuth.ClientCertificate,
+			"client_key":             cluster.MasterAuth.ClientKey,
+			"cluster_ca_certificate": cluster.MasterAuth.ClusterCaCertificate,
+		},
+	}
+	d.Set("master_auth", masterAuth)
+
+	d.Set("initial_node_count", cluster.InitialNodeCount)
+	d.Set("node_version", cluster.CurrentNodeVersion)
```

Comment:
As mentioned via IRC:

I tried testing the update, but there doesn't seem to be any way on how to create a cluster with older version of API, so I could test the upgrade...
either way this naming convention sits between "CurrentNodeVersion" & "DesiredNodeVersion" in the update structure, that's why I decided for `"node_version"`.

Do you still want me to rename this to `current_node_version`?